agent/agent_endpoint.go

```diff
@@ -1211,10 +1213,15 @@ func (s *HTTPHandlers) AgentMonitor(resp http.ResponseWriter, req *http.Request)
 			if droppedCount > 0 {
 				s.agent.logger.Warn("Dropped logs during monitor request", "dropped_count", droppedCount)
 			}
+			flusher.Flush()
 			return nil, nil
 		case log := <-logsCh:
 			fmt.Fprint(resp, string(log))
-			flusher.Flush()
+			select {
+			case <-t.C:
+				t = time.AfterFunc(flushDelay, flusher.Flush)
```

Comment:
`AfterFunc` calls `flusher.Flush` in a separate goroutine, which I'm not sure is necessarily safe. It also creates an unnecessary channel each time.

What do you think about using just a `time.Time` here?

```go
lastFlush := time.Now()

...

if time.Since(lastFlush) > flushDelay {
    flusher.Flush()
    lastFlush = time.Now()
}


```

Reply:
That is true, it does mean that when logs are infrequent we wouldn't get the last few logs until the monitor is closed. Since this is a debugging interface maybe that is ok? Generally a monitor is only going to be run for a few seconds or minutes. So even if we don't write logs often, when the user closes the monitor they should get the last few logs, right?

Reply:
The challenge is more with the tests, we will need to rework a lot of tests for this. As we use the log interceptor to check if we got the logs we expect and once we stop the monitor the interceptor is not anymore available to check the remaining of the logs.

Reply:
Added complexity is that I noticed the api tests fail on the pipeline but not locally when we miss/remove the flush. I think because locally we use localhost as interface so the flush is implicit.

Reply:
I changed it to use a ticker, it do not avoid using extra resources but it solve the race issue. IMO we should avoid having data not flushed until we close the connection. The user experience is not ideal.